src/main/java/org/deepsymmetry/beatlink/VirtualRekordbox.java

```diff
@@ -415,16 +421,38 @@ private DeviceUpdate buildUpdate(DatagramPacket packet) {
                         lastValidStatusFlagBytes.put(rawDeviceNumber, reportedStatusFlags);  // Record in case next packet for this device is malformed.
                     }
 
+                    // The raw ID sent from the Opus, we just need this to detect changes.
+                    // We can't use CdjStatus.getRekordboxId() because it still refers to the
+                    // playerToDeviceSqlRekordboxId map that hasn't been updated yet.
+                    int rawRekordboxId = (int)Util.bytesToNumber(packet.getData(), 0x2c, 4);
+
                     CdjStatus status = new CdjStatus(packet, hadToRecoverStatusFlags);
 
-                    // If source player number is zero the deck does not have a song loaded, clear the PSSI and source slot we had for that player.
-                    // This also represents a new track load.
-                    // We also need to set the DeviceSQL rekordbox ID to 0 for the player, so we don't
-                    // reflect the old track right when the new track is loaded (it takes a few seconds to match the PSSI).
-                    if (status.getTrackSourcePlayer() == 0) {
+                    // Get previous ID for comparison
+                    Integer previousId = previousRawRekordboxIds.get(status.getDeviceNumber());
+                    
+                    // Determine if track has changed and if it's currently loaded
+                    boolean isLoaded = status.getTrackSourcePlayer() != 0;
+                    boolean trackChanged = previousId != null && previousId != rawRekordboxId && rawRekordboxId != 0;
+                    
+                    // Clear slot and ID mapping if deck is empty or track has changed
+                    if (trackChanged || !isLoaded) {
                         playerTrackSourceSlots.remove(status.getDeviceNumber());
                         playerToDeviceSqlRekordboxId.remove(status.getDeviceNumber());
                     }
+                    
+                    // Only request PSSI when there's a new track loaded (ID change)
+                    if (trackChanged) {
+                        try {
+                            requestPSSI();
+                        } catch (IOException e) {
+                            logger.warn("Cannot send PSSI request");
+                        }
+                    }
+                    
+                    // Update the previous ID for next comparison
+                    previousRawRekordboxIds.put(status.getDeviceNumber(), rawRekordboxId);
```

Comment:
Note that you can do this up at line 432, because `put()` [returns the previous value](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/concurrent/ConcurrentHashMap.html#put(K,V)). So this line could be deleted, and line 432 could be:
```java
Integer previousId = previousRawRekordboxIds.put(status.getDeviceNumber(), rawRekordboxId);
```